src/CoreWCF.Http/tests/Async767311Tests.cs

```diff
@@ -98,20 +101,26 @@ internal void CallbackResults(IAsyncResult asyncResult)
         }
 
         [Fact]
-        public void Variation_CallbackMethod()
+        public async Task Variation_CallbackMethod()
         {
             IWebHost host = ServiceHelper.CreateWebHostBuilder<Startup>(_output).Build();
             using (host)
             {
-                host.Start();
+                await host.StartAsync();
                 System.ServiceModel.BasicHttpBinding httpBinding = ClientHelper.GetBufferedModeBinding();
                 var factory = new System.ServiceModel.ChannelFactory<IClientAsync_767311>(httpBinding, new System.ServiceModel.EndpointAddress(new Uri($"http://localhost:{host.GetHttpPort()}/BasicWcfService/SyncService.svc")));
                 IClientAsync_767311 clientAsync_ = factory.CreateChannel();
                 _output.WriteLine("Testing [Variation_CallbackMethod]");
                 AsyncCallback callback = new AsyncCallback(CallbackResults);
                 IAsyncResult result = clientAsync_.BeginEchoString(clientString, callback, null);
                 _output.WriteLine("Message sent via Async, waiting for callback");
-                autoEvent.WaitOne();
+                
+                bool signaled = autoEvent.WaitOne(TimeSpan.FromSeconds(30));
+                if (!signaled)
+                {
+                    Assert.True(false, "WaitOne timed out after 30 seconds");
+                }
```

Comment:
This is clumsy code. Call `Assert.True(signaled, ...)` instead of using Assert.True(false,...).